# Laboratorio 03: Fundamentos Estadísticos y Optimización

**Curso:** Inteligencia Artificial I – IS-484  
**Estudiante:** Sebastián Cahuin  
**Tema:** Análisis y predicción de matrículas y consumo eléctrico

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

matricula = pd.read_csv("matricula_sistemas.csv")
consumo = pd.read_csv("consumo_laboratorios.csv")

print("Datos de matrícula:")
display(matricula.head())

print("Datos de consumo:")
display(consumo.head())

Datos de matrícula:


,semestre,num_postulantes,becas_otorgadas,tasa_desercion_anterior,campania_difusion,num_matriculados
0,2015-I,211,10,0.106,0,120
1,2015-II,179,9,0.115,0,88
2,2016-I,180,12,0.150,0,101
3,2016-II,146,8,0.151,0,75
4,2017-I,198,11,0.158,1,112


Datos de consumo:


,num_equipos,horas_uso,consumo_kwh
0,29,5.1,19.6
1,8,6.8,13.3
2,13,3.5,9.8
3,5,7.1,16.0
4,26,6.3,19.7


In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
from sklearn.linear_model import LinearRegression

matricula = pd.read_csv("matricula_sistemas.csv")
consumo = pd.read_csv("consumo_laboratorios.csv")

print("Datos de matrícula:")
display(matricula.head())

print("Datos de consumo:")
display(consumo.head())

Datos de matrícula:


,semestre,num_postulantes,becas_otorgadas,tasa_desercion_anterior,campania_difusion,num_matriculados
0,2015-I,211,10,0.106,0,120
1,2015-II,179,9,0.115,0,88
2,2016-I,180,12,0.150,0,101
3,2016-II,146,8,0.151,0,75
4,2017-I,198,11,0.158,1,112


Datos de consumo:


,num_equipos,horas_uso,consumo_kwh
0,29,5.1,19.6
1,8,6.8,13.3
2,13,3.5,9.8
3,5,7.1,16.0
4,26,6.3,19.7


In [2]:
matricula.describe()

,num_postulantes,becas_otorgadas,tasa_desercion_anterior,campania_difusion,num_matriculados
count,24.000000,24.000000,24.000000,24.000000,24.000000
mean,231.125000,18.250000,0.129917,0.500000,137.083333
std,41.277757,6.917212,0.021055,0.510754,34.122435
min,146.000000,8.000000,0.096000,0.000000,75.000000
25%,200.250000,12.750000,0.110750,0.000000,111.500000
50%,222.500000,17.500000,0.127500,0.500000,137.000000
75%,261.250000,24.250000,0.150250,1.000000,161.500000
max,311.000000,30.000000,0.167000,1.000000,198.000000


In [3]:
col = matricula["num_matriculados"]

print("Media:", col.mean())
print("Mediana:", col.median())
print("Moda:", col.mode().tolist())
print("Rango:", col.max() - col.min())
print("Varianza:", col.var())
print("Desviación estándar:", col.std())

Media: 137.08333333333334
Mediana: 137.0
Moda: [120, 137, 160]
Rango: 123
Varianza: 1164.340579710145
Desviación estándar: 34.12243513745971


La matrícula tiene un promedio de 137.08 estudiantes por semestre y una mediana de 137. La moda es múltiple: 120, 137 y 160 son los valores que más se repiten. El rango es de 123 estudiantes, lo que muestra una diferencia considerable entre el mínimo y el máximo. La desviación estándar es de aproximadamente 34.12, por lo que la matrícula suele variar alrededor de 34 estudiantes respecto al promedio.